external-crates/move/crates/move-stackless-bytecode/src/conditions_order_analysis.rs

```diff
@@ -0,0 +1,376 @@
+        let func_target = FunctionTarget::new(func_env, &data);
+
+        let underlying_func_id = targets
+            .function_specs()
+            .iter()
+            .find_map(|(id, func)| (func_env.get_qualified_id() == *id).then_some(*func));
+
+        if underlying_func_id.is_none() {
+            env.diag(
+                Severity::Error,
+                &func_env.get_loc(),
+                "Spec underlying func is not found",
+            );
+
+            return  data;
+        }
+
+        let underlying_func = func_env.module_env.get_function(underlying_func_id.unwrap().id);
+
+        let spec_params = func_env.get_parameters();
+        let underlying_params = underlying_func.get_parameters();
+
+        if spec_params.len() != underlying_params.len() {
+            env.diag(
+                Severity::Error,
+                &func_env.get_loc(),
+                "Spec function have differ params count than underlying func",
+            );
+
+            return data;
+        }
+
+        for i in 0..spec_params.len() {
+            if spec_params[i].0 != underlying_params[i].0 {
+                env.diag(
+                    Severity::Warning,
+                    &func_env.get_loc(),
+                    "Spec function have differ params names than underlying func",
+                );
+            }
+
+            if spec_params[i].1 != underlying_params[i].1 {
+                env.diag(
+                    Severity::Error,
+                    &func_env.get_loc(),
+                    "Spec function have differ params type than underlying func",
+                );
+
+                return data;
+            }
+        }
+
+        let code = func_target.get_bytecode();
+        let cfg: StacklessControlFlowGraph = StacklessControlFlowGraph::new_forward(code);
+        let entry = cfg.entry_block();
+        let nodes = cfg.blocks();
+        let edges: Vec<(BlockId, BlockId)> = nodes
+            .iter()
+            .flat_map(|x| {
+                cfg.successors(*x)
+                    .iter()
+                    .map(|y| (*x, *y))
+                    .collect::<Vec<(BlockId, BlockId)>>()
+            })
+            .collect();
+        let graph: Graph<u16> = Graph::new(entry, nodes, edges);
+        let builder = FunctionDataBuilder::new(&func_env, data.clone());
+
+        let (call_node_id, call_operation, multiple_calls) = self.find_node_by_func_id(underlying_func.get_qualified_id(), &graph, code, &cfg);
+
+        if !call_node_id.is_some() {
+            env.diag(
+                Severity::Error,
+                &func_env.get_loc(),
+                "Consider add function call to spec",
+            );
+
+            return data;
+        }
+
+        if multiple_calls {
+            env.diag(
+                Severity::Error,
+                &func_env.get_loc(),
+                "Underlying func is calling few times",
+            );
+
+            return data;
+        }
+
```

Comment:
we need a check here that the arguments and type arguments  of `call_operation` are the same as the parameters and type parameters of the spec function.